presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -82,6 +85,7 @@
     private final int maxPartitionBatchSize;
     private final int maxInitialSplits;
     private final boolean recursiveDfsWalkerEnabled;
+    private final CounterStat highMemorySplitSourceCounter;
```

Comment:
Why not initialize this here?

Reply:
Both constructors in HiveSplitManager are public. When the second one is used, this counter is passed in from outside.